src/Cassandra/Data/Linq/CqlQueryExtensions.cs

```diff
@@ -95,6 +106,34 @@ public static CqlQuery<TSource> Where<TSource>(this CqlQuery<TSource> source, Ex
         }
 
         /// <summary>
+        ///  Returns a CqlQuery which after execution returns grouped sequence of values based on a predicate.
+        ///  To execute this CqlQuery use <c>Execute()</c> method.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements of source.</typeparam>
+        /// <typeparam name="TKey">The type of the value returned by selector.</typeparam>
+        /// <param name="source">The CqlQuery&lt;TSource&gt; to filter.</param>
+        /// <param name="predicate">A function to test each element for a condition.</param>
+        /// <returns>a CqlQuery&lt;TSource&gt; which after execution will return an IEnumerable&lt;TSource&gt;
+        /// that contains elements from the input sequence that satisfy the condition.</returns>
+        public static CqlQuery<IGrouping<TKey, TSource>> GroupBy<TKey, TSource>(this CqlQuery<TSource> source, 
```

Comment:
I see what you mean... if `ICassandraGrouping` inherited from `IGrouping` the extension methods would still apply, if `ICassandraGrouping` didn't inherit it would be a hell of an effort to implement those methods...